src/deploy-config.sh

```diff
@@ -2,41 +2,91 @@
 set -euo pipefail
 IFS=$'\n\t'
 
-subscriptionId=$1
-resourceGroupName=$2
-nginxDeploymentName=$3
-nginxConfigurationFile=$4
+subscription_id=$1
+resource_group_name=$2
+nginx_deployment_name=$3
+config_dir_path=$4
+root_config_file_name=$5
+transformed_config_dir_path=${6:-''}
 
-# Read and encode the NGINX configuration file content.
-if [ -f "$nginxConfigurationFile" ]
+# Validation and preprocessing
+
+if [[ "$config_dir_path" = /* ]]
+then
+    echo "The NGINX configuration directory path in the repository '$config_dir_path' must be a relative path."
+    exit 2
+elif [[ ! "$config_dir_path" = */ ]]
+then
+    echo "The NGINX configuration directory path '$config_dir_path' does not end with '/'. Appending a trailing '/'."
+    config_dir_path="$config_dir_path/"
+fi
+
+if [[ -d "$config_dir_path" ]]
+then
+    echo "The NGINX configuration directory '$config_dir_path' was found."
+else 
+    echo "The NGINX configuration directory '$config_dir_path' does not exist."
+    exit 2
+fi
+
+root_config_file_path="$config_dir_path$root_config_file_name"
+if [[ -f "$root_config_file_path" ]]
 then
-    echo "The NGINX configuration file was found."
+    echo "The root NGINX configuration file '$root_config_file_path' was found."
 else 
-    echo "The NGINX configuration file $nginxConfigurationFile does not exist."
+    echo "The root NGINX configuration file '$root_config_file_path' does not exist."
     exit 2
 fi
 
-encodedConfigContent=$(base64 "$nginxConfigurationFile")
-echo "Base64 encoded NGINX configuration content"
-echo "$encodedConfigContent"
+if [[ -n "$transformed_config_dir_path" ]]
+then
+    if [[ ! "$transformed_config_dir_path" = /* ]]
+    then
+        echo "The specified transformed NGINX configuration directory path '$transformed_config_dir_path' must be an absolute path that starts with '/'."
```

Comment:
Why does  this have to be absolute?

Reply:
It doesn't have to be as this string just replaces the leading "./" from the relative file paths tar sees.
On the other hand, is there a use case for overwriting the relative path? If the customer NGINX config relies on relative paths for include, will ignore this parameter just work for them?

Reply:
I don't have a use case, but the assertion that it _must_ be absolute stood out to me as I couldn't think of a reason other than it being the more likely case where you would want to transform paths.

Reply:
If there is no use case for transforming a relative path, does artificially limiting its usage (by validation) to absolute path help reduce user errors of using it? Basically, I feel it is clearer to convey to users that use this transformed path if your NGINX config files rely on include absolute paths, otherwise leave this option unused.

Reply:
> does artificially limiting its usage (by validation) to absolute path help reduce user errors of using it?

If you think it does then that's the reason to enforce it and the answer to my question.

Reply:
Sounds good!